libs/@local/graph/postgres-store/src/store/postgres/ontology/property_type.rs

```diff
@@ -295,96 +297,125 @@ where
+                                    BorrowedTraversalParams::ResolveDepths {
+                                        graph_resolve_depths: new_graph_resolve_depths,
+                                    },
                                     traversal_interval,
                                 );
                             }
                         }
                     }
-                    SubgraphTraversalParams::Paths { traversal_paths: _ } => todo!("https://linear.app/hash/issue/BE-103/implement-path-based-query-traversal-depths"),
+                    BorrowedTraversalParams::Path { traversal_path } => {
+                        let Some((edge, rest)) = traversal_path.split_first() else {
+                            continue;
+                        };
+
+                        let edge_kind = match edge {
+                            TraversalEdge::ConstrainsPropertiesOn {
+                                direction: OntologyTraversalEdgeDirection::Outgoing,
+                            } => OntologyEdgeKind::ConstrainsPropertiesOn,
+                            TraversalEdge::ConstrainsValuesOn {
+                                direction: OntologyTraversalEdgeDirection::Outgoing,
+                            } => OntologyEdgeKind::ConstrainsValuesOn,
+                            TraversalEdge::InheritsFrom { .. }
+                            | TraversalEdge::ConstrainsLinksOn { .. }
+                            | TraversalEdge::ConstrainsLinkDestinationsOn { .. }
+                            | TraversalEdge::IsOfType { .. }
+                            | TraversalEdge::HasLeftEntity { .. }
+                            | TraversalEdge::HasRightEntity { .. } => continue,
+                        };
+
+                        edges_to_traverse.entry(edge_kind).or_default().push(
+                            OntologyTypeUuid::from(property_type_ontology_id),
+                            BorrowedTraversalParams::Path {
+                                traversal_path: rest,
+                            },
+                            traversal_interval,
+                        );
+                    }
                 }
             }
 
-            if let Some(traversal_data) =
-                edges_to_traverse.get(&OntologyEdgeKind::ConstrainsValuesOn)
-            {
-                data_type_queue.extend(
-                    Self::filter_data_types_by_permission(
-                        self.read_ontology_edges::<PropertyTypeVertexId, DataTypeVertexId>(
-                            traversal_data,
-                            ReferenceTable::PropertyTypeConstrainsValuesOn,
-                        )
-                        .await?,
-                        provider,
-                        subgraph.temporal_axes.resolved.clone(),
+            for (edge_kind, table) in [(
+                OntologyEdgeKind::ConstrainsValuesOn,
+                ReferenceTable::PropertyTypeConstrainsValuesOn,
+            )] {
```

Comment:
This loop seems a bit suspect to me

Reply:
It's just for consistency. The pattern is the same everywhere. No strong opinion to keep it or not